src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -545,27 +547,72 @@ Namespace Microsoft.CodeAnalysis.Semantics
             End If
 
             Dim operand As Lazy(Of IOperation)
+            Dim type As ITypeSymbol = boundConversion.Type
+            Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundConversion.ConstantValueOpt)
+            Dim isImplicit As Boolean = boundConversion.WasCompilerGenerated
             Dim methodSymbol As MethodSymbol
 
             If (boundConversion.ConversionKind And VisualBasic.ConversionKind.UserDefined) = VisualBasic.ConversionKind.UserDefined Then
                 Dim userDefinedConversion As BoundUserDefinedConversion = DirectCast(boundConversion.Operand, BoundUserDefinedConversion)
                 methodSymbol = userDefinedConversion.Call.Method
                 operand = New Lazy(Of IOperation)(Function() Create(userDefinedConversion.Operand))
+            ElseIf boundConversion.ConversionKind = VisualBasic.ConversionKind.Identity AndAlso
+                   boundConversion.Operand.Kind = BoundKind.DelegateCreationExpression Then
+                ' In this scenario, we're something like New DelegateType(AddressOf Method)
+                ' We don't want to process both this conversion and the DelegateCreationExpression
```

Comment:
>We don't want to process both this conversion and the DelegateCreationExpression [](start = 18, length = 80)

This is probably true only if the conversion is implicit?

Reply:
Could you point me to the test that is affected by this logic? 

---
In reply to: [138492959](https://github.com/dotnet/roslyn/pull/22056#discussion_r138492959) [](ancestors = 138492959)

Reply:
I believe here and below we should have more targeted logic, let's discuss offline.

---
In reply to: [138494022](https://github.com/dotnet/roslyn/pull/22056#discussion_r138494022) [](ancestors = 138494022,138492959)

Reply:
I'll be in the office later today, I'll stop by then. As an example though, https://github.com/dotnet/roslyn/pull/22056/files#diff-076069470e21e72cef89138ff31fa9f4R658. We create a BoundConversion on top of a BoundDelegateCreation in the case of `New DelegateType(AddressOf Method)`. From my reading of the spec, this should be exactly equivalent to `AddressOf Method`, so we shouldn't have nested `IDelegateCreationExpression`s, like we do for `new Action((Action)Method)` in C#.